lib/doorkeeper/config/option.rb

```diff
@@ -35,17 +35,24 @@ module Option
       def option(name, options = {})
         attribute = options[:as] || name
         attribute_builder = options[:builder_class]
+        deprecated = options[:deprecated]
 
         Builder.instance_eval do
           remove_method name if method_defined?(name)
-          define_method name do |*args, &block|
-            value = if attribute_builder
-                      attribute_builder.new(&block).build
-                    else
-                      block || args.first
-                    end
+          if deprecated
```

Comment:
Don't see any reason to store options value in a local var here because we use it only one time, so maybe directly here? (`if options[:deprecated]`)

Reply:
True. Fixed.